src/org/usfirst/frc4904/robot/humaninterface/drivers/DriveStraightSpeedModifier.java

```diff
@@ -0,0 +1,45 @@
+package org.usfirst.frc4904.robot.humaninterface.drivers;
+
+
+import org.usfirst.frc4904.standard.LogKitten;
+import org.usfirst.frc4904.standard.Util;
+import org.usfirst.frc4904.standard.custom.ChassisController;
+import org.usfirst.frc4904.standard.custom.motioncontrollers.MotionController;
+import org.usfirst.frc4904.standard.custom.sensors.InvalidSensorException;
+import org.usfirst.frc4904.standard.subsystems.motor.speedmodifiers.SpeedModifier;
+
+public class DriveStraightSpeedModifier implements SpeedModifier {
+	protected final ChassisController controller;
+	protected final MotionController motionController;
+
+	public DriveStraightSpeedModifier(ChassisController controller, MotionController motionController) {
+		this.controller = controller;
+		this.motionController = motionController;
+	}
+
+	@Override
+	public double modify(double speed) {
+		// If we're driving straight, use drive straight PID
+		if (Util.isZero(controller.getY()) && Util.isZero(controller.getX())) {
+			// Tell the motion controller to hold the current heading
+			motionController.setSetpoint(motionController.getSensorValue());
```

Comment:
Is there a particular reason that this happens before enabling the controller? Just curious, not actually a problem